src/server/mod.rs

```diff
@@ -53,11 +63,230 @@ where
+                                Err(err) if err.is_disconnected() => (),
+                                Err(err) => unreachable!("Unexpected channel err {}", err),
+                            }
+                            
+                            return;
+                        }
+                        Ok(n) => {
+                            pos += n;
+                            n
+                        }
+                        Err(err) => {
+                            match io_primary_send.send(Err(err)).await {
+                                Ok(_) => (),
+                                Err(err) if err.is_disconnected() => (),
+                                Err(err) => unreachable!("Unexpected channel err {}", err),
+                            }
+                            return
+                        }
+                    };
+                    
+                    if pos < 3 {
+                        continue
+                    }
+                    
+                    for idx in 3..n {
+                        if idx >= 3 && &buf[idx - 3..=idx] == b"\r\n\r\n" {
+                            break 'a idx;
+                        }
+                    }
+                    
+                    if pos == buf.len() {
+                        let res = Ok((Arc::new(vec![].into_boxed_slice()), 0usize));
+                        match io_primary_send.send(res).await {
+                            Ok(_) => (),
+                            Err(err) if err.is_disconnected() => (),
+                            Err(err) => unreachable!("Unexpected channel err {}", err),
+                        }
+                        error!("Headers exceeded 8kb, rejecting request");
+                        return
+                    }
+                };
+                
+                let body_start = buf.split_off(header_end_pos+1);
+                
+                let res = Ok((Arc::new(buf.into_boxed_slice()), header_end_pos+1));
+                match io_primary_send.send(res).await {
+                    Ok(_) => (),
+                    Err(err) if err.is_disconnected() => return,
+                    Err(err) => unreachable!("Unexpected channel err {}", err),
+                }
+                
+                let body_start_n = pos - (header_end_pos+1);
+                
+                (body_start, body_start_n)
+            };
+            
+            let mut read_failure = None;
+            
+            // read more if no body has been recieved
+            if n == 0 {
+                // use a smaller buffer than used to read headers
+                // to avoid resize in most cases
+                if body_start.len() < 6 * 1024 {
+                    body_start.resize(8 * 1024, 0);
+                }
+                
+                n = match _io.read(&mut body_start).await {
+                    Ok(n) => {
+                        n
+                    }
```

Comment:
```suggestion
                    Ok(n) => n,
```

Reply:
in general, this needs rustfmt run on the whole pr. i've mentioned this before to @Yarn